pac4j-core/src/main/java/org/pac4j/core/credentials/extractor/BasicAuthExtractor.java

```diff
@@ -28,26 +29,26 @@ public BasicAuthExtractor(final String headerName, final String prefixHeader) {
     }
 
     @Override
-    public UsernamePasswordCredentials extract(WebContext context) {
-        final TokenCredentials credentials = this.extractor.extract(context);
-        if (credentials == null) {
-            return null;
-        }
-
-        final byte[] decoded = Base64.getDecoder().decode(credentials.getToken());
-
-        String token;
-        try {
-            token = new String(decoded, "UTF-8");
-        } catch (final UnsupportedEncodingException e) {
-            throw new CredentialsException("Bad format of the basic auth header");
-        }
-
-        final int delim = token.indexOf(":");
-        if (delim < 0) {
-            throw new CredentialsException("Bad format of the basic auth header");
-        }
-        return new UsernamePasswordCredentials(token.substring(0, delim),
-                token.substring(delim + 1));
+    public Optional<UsernamePasswordCredentials> extract(WebContext context) {
+        final Optional<TokenCredentials> credentials = this.extractor.extract(context);
```

Comment:
do not use an intermediary variable, directly go for this.extractor.extract(context).map ...